File: app/src/main/java/me/imzack/app/end/injector/module/TypeSearchPresenterModule.java
package me.imzack.app.end.injector.module;

import me.imzack.app.end.view.contract.TypeSearchViewContract;

import dagger.Module;
import dagger.Provides;

@Module
public class TypeSearchPresenterModule {

    private final TypeSearchViewContract mTypeSearchViewContract;

    public TypeSearchPresenterModule(TypeSearchViewContract typeSearchViewContract) {
        mTypeSearchViewContract = typeSearchViewContract;
    }

    @Provides
    TypeSearchViewContract provideTypeSearchViewContract() {
        return mTypeSearchViewContract;
    }
}

File: app/src/main/kotlin/me/imzack/app/end/injector/module/TypeSearchPresenterModule.kt
package me.imzack.app.end.injector.module

import dagger.Module
import dagger.Provides
import me.imzack.app.end.view.contract.TypeSearchViewContract

@Module
class TypeSearchPresenterModule(private val mTypeSearchViewContract: TypeSearchViewContract) {

    @Provides
    fun provideTypeSearchViewContract() = mTypeSearchViewContract
}
